core/Tracker/GoalManager.php

```diff
@@ -677,6 +678,12 @@ protected function recordStandardGoals(VisitProperties $visitProperties, Request
             $conversionDimensions = ConversionDimension::getAllDimensions();
             $conversion = $this->triggerHookOnDimensions($request, $conversionDimensions, 'onGoalConversion', $visitor, $action, $conversion);
 
+            if ($this->isEventMatchingGoal($convertedGoal)
+                && !empty($convertedGoal['event_value_as_revenue'])
+            ) {
+                $conversion['revenue'] = ActionEvent::getEventValue($request);
```

Comment:
@diosmosis just checking: wouldn't this always overwrite the default revenue? Even if eventValue is `0` for example or none set?

Reply:
Good point, will fix w/ test.